src/index.ts

```diff
@@ -1,25 +1,475 @@
+    repo: userConfig.repo,
+    pull_number: prNum
+  })
+  return res;
+}
+
+async function getSingleCommit(client: Octokit, sha: any) {
+  const res = await client.repos.getCommit({
+    owner: userConfig.owner,
+    repo: userConfig.repo,
+    sha: sha
+  })
+  return res;
+}
+
+async function getSingleIssue(client: Octokit, issueNum: number) {
+  const res = await client.issues.get({
+    owner: userConfig.owner,
+    repo: userConfig.repo,
+    issue_number: issueNum
+  })
+  return res;
+}
+
+// create a label for this repo, can be duplicated
+async function createLabel(client: Octokit, name: string, col: string) {
+  const res = await client.issues.createLabel({
+    owner: userConfig.owner,
+    repo: userConfig.repo,
+    name: name,
+    color: col
+  })
+  return res;
+}
+
+//events: pinned, unpinned, labeled, unlabeled, etc.
+async function listEventForIssue(client: Octokit, issueNum: number) {
+  const res = await client.issues.listEvents({
+    owner: userConfig.owner,
+    repo: userConfig.repo,
+    issue_number: issueNum
+  })
+  return res;
+}
+
+async function listCommitsForPR(client: Octokit, pullNum: number) {
+  const res = await client.pulls.listCommits({
+    owner: userConfig.owner,
+    repo: userConfig.repo,
+    pull_number: pullNum,
+    per_page: 100,
+    page: 1
+  });
+  return res;
+}
+
+async function addLabelsToIssue(client: Octokit, issueNum: number, labels: Array<string>) {
+  const res = await client.issues.addLabels({
+    owner: userConfig.owner,
+    repo: userConfig.repo,
+    issue_number: issueNum,
+    labels: labels
+  })
+  return res;
+}
+
+async function deleteLabel(client: Octokit, name: string) {
+  const res = await client.issues.deleteLabel({
+    owner: userConfig.owner,
+    repo: userConfig.repo,
+    name: name
+  })
+  return res;
+}
+
+function isGreaterThan(end: number, begin: number, unit: string, times: string) {
+  if (unit === "") {
+    return (end - begin) >= Number(times)
+  }
+  switch (unit) {
+    case 'weeks':
+      return (end - begin) >= WEEK_MM * Number(times)
+    case 'days':
+      return (end - begin) >= DAY_MM * Number(times)
+    case 'hours':
+      return (end - begin) >= HR_MM * Number(times)
+    default:
+      return true;
+  }
+}
+
+// go through all labels, check if there is label with prefix, if lableName is empty, check if there is any label
+function hasLabelWithPrefix(issue: any, prefix: String) {
+  const labelList = issue.labels;
+  if (isEmpty(labelList)) {
+    return false;
+  }
+
+  labelList.forEach((label: any) => {
+    const name = label.name;
+    if (name.startsWith(prefix)) {
+      return true;
+    }
+  });
+  return false;
+}
+
+//checkk if the pr contains commit with 'cherry picked from commit xxxx'
+function IsPRContainsCherryPickCommit(pr: any) {
+  listCommitsForPR(client, pr.number).then(response => {
+    response.data.forEach(eachCom => {
+      if (includes(eachCom.commit.message, 'cherry picked from commit')) {
```

Comment:
This string could be a constant in the configuration file if it's possible it's different. Either way, since it's used multiple time in this function we should assign it to a const at the start of the function.